### PR TITLE
Add ManifestBuildData

### PR DESCRIPTION
**Pull request by chcosta**
Fixes https://github.com/dotnet/arcade/issues/2150

Validation is in progress

FYI @mmitche @jcagme

### CODE REVIEW

**Comment by chcosta** [comment]
Yes, but why can't we just auto-add it based on `ExpectedFeedUrl`?  It seems redundant to require both.

**Comment by mmitche** [comment]
Not sure? I suppose that's one option.  tbh I can't remember why it was done this way vs. directly off of the ExpectedFeedUrl.  @jcagme may remember.

**Comment by chcosta** [comment]
I've validated that this now adds "Location" if you don't specify it in "ManifestBuildData".  @jcagme, are you ok with this change?  

@mmitche , take another look?

Internal build - https://dnceng.visualstudio.com/internal/_build/results?buildId=113604

**Comment by chcosta** [comment]
This seems to be publishing the manifest from the second push that the internal build does.  I'll try to get this fixed but it's a bit of ugliness

**Comment by chcosta** [comment]
Internal build - https://dev.azure.com/dnceng/internal/_build/results?buildId=115258

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs [hunk 1]
```diff
@@ -153,6 +153,12 @@ private async Task<bool> ExecuteAsync()
                     blobArtifacts = ConcatBlobArtifacts(blobArtifacts, symbolItems);
                 }
 
+                if(!BuildManifestUtil.ManifestBuildDataHasLocationProperty(ManifestBuildData))
+                {
+                    string[] locationAttribute = new string[] { $"Location={ExpectedFeedUrl}" };
+                    ManifestBuildData = ManifestBuildData == null ? locationAttribute: ManifestBuildData.Concat( locationAttribute ).ToArray();
```
Comment (jcagme) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs]: > locationAttribute  [](start = 112, length = 19)

Nit: remove spaces

**Comment by jcagme** [comment]
Just a minor nit, other than that LGTM

**Comment by chcosta** [comment]
FYI @JohnTortugo , as this makes changes to validate-sdk.ps1 which may or may not be relevant to your work.

**Pull request merged**